adapter-pinecone/src/main/java/io/nosqlbench/adapter/pinecone/opdispensers/PineconeDeleteOpDispenser.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.adapter.pinecone.opdispensers;
+
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import io.nosqlbench.adapter.pinecone.PineconeDriverAdapter;
+import io.nosqlbench.adapter.pinecone.PineconeSpace;
+import io.nosqlbench.adapter.pinecone.ops.PineconeDeleteOp;
+import io.nosqlbench.adapter.pinecone.ops.PineconeOp;
+import io.nosqlbench.engine.api.templating.ParsedOp;
+import io.pinecone.proto.DeleteRequest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.*;
+import java.util.function.LongFunction;
+
+
+public class PineconeDeleteOpDispenser extends PineconeOpDispenser {
+    private static final Logger LOGGER = LogManager.getLogger(PineconeDeleteOpDispenser.class);
+    private final LongFunction<DeleteRequest> deleteRequestFunc;
+
+    /**
+     * Create a new PineconeDeleteOpDispenser subclassed from {@link PineconeOpDispenser}.
+     *
+     * @param adapter           The associated {@link PineconeDriverAdapter}
+     * @param op                The {@link ParsedOp} encapsulating the activity for this cycle
+     * @param pcFunction        A function to return the associated context of this activity (see {@link PineconeSpace})
+     * @param targetFunction    A LongFunction that returns the specified Pinecone Index for this Op
+     */
+    public PineconeDeleteOpDispenser(PineconeDriverAdapter adapter,
+                                     ParsedOp op,
+                                     LongFunction<PineconeSpace> pcFunction,
+                                     LongFunction<String> targetFunction) {
+        super(adapter, op, pcFunction, targetFunction);
+        deleteRequestFunc = createDeleteRequestFunction(op);
+    }
+
+    @Override
+    public PineconeOp apply(long value) {
+        return new PineconeDeleteOp(pcFunction.apply(value).getConnection(targetFunction.apply(value)),
```

Comment:
nit: Might be easier to put these args in block form for readability

Reply:
Done